docs/lsm/partial_compiles/partial_compiles.rst

```diff
@@ -0,0 +1,117 @@
+Partial Compiles
+****************
+
+
+Partial compilation is an approach to speed up compilation when the Service Inventory contains many instances.
+
+Ordinarily, LSM re-compiles all instances on every update. This means that as the inventory grows, the compiles become slower. Partial compiles allow LSM to re-compile only those instances that are relevant to the current service instance, avoiding any slowdown. 
+
+Supported scenarios
+-------------------
+
+Partial compiles are possible when
+
+1. Service Instances are unrelated: service instances don't share any resources and don't depend on each other in any way. This requires no modifications to the model.
+2. Services form groups under a common owner. 
+    Instances within the group can freely depend on each other and share resources, but nothing is shared across group. 
+    One specific instance is designated as the owner of the group.
+    This requires indicating what the parent of any service is, by setting :inmanta:relation:`lsm::ServiceEntityBinding.owner` and :inmanta:attribute:`lsm::ServiceEntityBinding.relation_to_owner`.
+
+3. Service instances and groups can depend on shared resources, that are identical for all service instances and groups. This requires no modifications to the model.
+4. Any combination of the above
+
+Example
+-------------------
+
+As an example, consider the following model for managing ports and routers.
+Both are independent services, but a port can only be managed in combination with its router and all its sibling. 
+(This is not in general true, we often managed port without managing the entire router, but we use it as an example.)
+
+This model is not much different from normal :ref:`Inter Service Relations<inter_service_relations>`, except for lines 55-56.
+
+.. literalinclude:: partial.cf
+    :linenos:
+    :language: inmanta
+    :emphasize-lines: 55-56
+    :caption: main.cf
+
+
+How it works
+-------------------
+
+There are two things to consider:
+1. how to divide the resources into resource sets
+2. how to get the correct instances into the model
+
+Resource sets
++++++++++++++
+The key mechanism behind partial compiles are ``ResourceSet``: all resources in the desired state are divided into groups.
+When building a new desired state, instead of replacing the entire desired state, we only replace specific ``ResourceSet``.
+Resources in ``ResourceSet`` can not depend on Resources in other ``ResourceSets``.
```

Comment:
```suggestion
The key mechanism behind partial compiles are ``ResourceSets``: all resources in the desired state are divided into groups.
When building a new desired state, instead of replacing the entire desired state, we only replace a specific ``ResourceSet``.
Resources in a ``ResourceSet`` can not depend on Resources in other ``ResourceSets``.
```

Reply:
I think you forgot to apply this change.